### PR TITLE
request: development <- 67 api default application responses

### PR DESCRIPTION
**Pull request by Cusatelli**

### CODE REVIEW

File: src/main/java/com/noroff/mefit/data/service/GoalService.java [hunk 1]
```diff
@@ -1,40 +1,102 @@
...
+        Goal goal = Objects.requireNonNull(getById(goalId).getBody()).getPayload();
+        if(goal == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(
+                    new DefaultResponse<>(HttpStatus.FOUND.value(), DefaultResponse.FOUND(TAG, goalId))
+            );
+        }
+
+        goal.getWorkouts().remove(workout);
+        return this.update(goalId, goal);
     }
```
Comment (Cusatelli) [inline src/main/java/com/noroff/mefit/data/service/GoalService.java:101]: I think this is unused.
```suggestion
```

File: src/main/java/com/noroff/mefit/data/service/ProgramService.java [hunk 2]
```diff
@@ -1,49 +1,101 @@
...
+        Program program = Objects.requireNonNull(getById(programId).getBody()).getPayload();
+        if(program == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(
+                    new DefaultResponse<>(HttpStatus.FOUND.value(), DefaultResponse.FOUND(TAG, programId))
+            );
+        }
+
+        program.getWorkouts().remove(workout);
+        return this.update(programId, program);
     }
```
Comment (Cusatelli) [inline src/main/java/com/noroff/mefit/data/service/ProgramService.java:100]: Same here
```suggestion
```

**Review by OmarAbdiAli — APPROVED** [review]
Nice!

**Review by Cusatelli — COMMENTED** [review]
Maybe check out these unused methods?

**Pull request merged**